odl/trafos/wavelet.py

```diff
@@ -750,11 +828,26 @@ def __init__(self, ran, nscales, wbasis, mode):
 
             'per': periodization -- like periodic-padding but gives the
             smallest possible number of decomposition coefficients.
+
+            With J.-O. Stomberg's biorthogonal wavelets only signal
+            extension mode 'sym' is available.
         """
         self.nscales = int(nscales)
-        self.wbasis = wbasis
         self.mode = str(mode).lower()
 
+        if isinstance(wbasis, pywt.Wavelet):
+            self.wbasis = wbasis
+        else:
+            try:
+                wbasis = wbasis + ''
+                if wbasis.startswith('JOS'):
+                    self.wbasis = wbasis
```

Comment:
Here `self.wbasis` can be either a `str` or a `pywt.Wavelet` such type mixings are in general not good since any consumer of this property would need to check which one it is at all times.